w11_d1_1_node_express_mongo_review/README.md

```diff
@@ -0,0 +1,313 @@
+  Question.find({}, function (err, questions) {
+    res.json(questions);
+  });
+});
+```
+
+Remember, the `app.VERB` methods in our Express code say what the server should do when it receives a request, including what kind of response it should send back.  Here, we use `Question.find` (from mongoose and our `question.js` file) to find all the questions in the databse. The empty object we gave the query as the first argument means that the questions won't be filtered down at all before they're sent back. In the callback from the database query we send all the questions back to the client as json with `res.json`.  Note that we didn't put in any way to handle a database error if one occurs.
+
+
+
+```js
+
+// server.js
+
+// create new question
+app.post('/questions', function (req, res) {
+  // create new question with data from the body of the request (`req.body`)
+  // body should contain the question text itself
+  var newQuestion = new Question({
+    text: req.body.text
+  });
+
+  // save new question
+  newQuestion.save(function (err, savedQuestion) {
+    res.json(savedQuestion);
+  });
+});
+```
+
+Here we use mongoose for database interactions, again. The `new` method creates a new document, and `save` actually adds it to our database. We also incorporate data from the request, using `req.body`. We have easy access to data from the request body because of `body-parser`.
+
+```js
+// server.js
+
+// update question, but only the part(s) passed in in the request body
+// not currently that exciting when question has only one attribute
+app.put('/questions/:id', function (req, res) {
+  // set the value of the id
+  var targetId = req.params.id;
+
+  // find question in db by id
+  Question.findOne({_id: targetId}, function (err, foundQuestion) {
+    // update the question's text, if the new text passed in was truthy
+    // otherwise keep the same text
+    foundQuestion.text = req.body.text || foundQuestion.text;
+
+    // save updated question in db
+    foundQuestion.save(function (err, savedQuestion) {
+      res.json(savedQuestion);
+    });
+  });
+});
+```
+
+The final additional feature to show here is the addition of an `id` that comes to our server as a url parameter and corresponds to the database id of the question to edit. We take it out of the request with `req.params.id`, and we use `find` _with parameters_, `{_id: targetId}` to find the desired question in the database. As with creating a new post, we have to `save` to actually put changes into the database.
+
+## Challenges
+
+1. Test your current routes using Postman.
+2. Create a few questions in your database with Postman.
+1. In `server.js`, add the correct restful route method declarations for the actions listed below. For this step, just get the routes set up and use comments to plan what should happen in each route.
+  * Get one specific question.
+  * Delete one specific question.
+  * Update a question by completely overwriting its data.
```

Comment:
I like the distinction between this type of update, and the one you show above (checking that the new data is truthy). I think it would help to add a short explanation on why we'd want to completely override its data instead.